BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Gets a device from the "devices" collection, given an objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: BsonDocument that was requested
+         */
+        public List<BsonDocument> GetAllDevices()
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Empty;
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var docs = devices.Find(filter).ToList();
+                return docs;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return null;
+            }
+        }
+
+        /*
+         * Deletes from the "devices" collection, given the objectID.
+         * Parameter: objectId, corresponds to the "_id" field for a given document inside of MongoDB
+         * Returns: true if successful, false if not successful
+         */
+        public bool DeleteDevice(ObjectId objectId)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            var filter = Builders<BsonDocument>.Filter.Eq("_id", objectId);
+
+            try
+            {
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var result = devices.DeleteOne(filter);
+                return result.DeletedCount > 0;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return false;
+            }
+        }
+
+        /*
+         * Updates a device in the "devices" collection, given a fully formed updated device.
+         * Parameters: BsonDocument containing the updated BsonDocument.
+         * Returns: true if successful, false if unsuccessful
+         */
+        public bool UpdateDevice(BsonDocument doc)
+        {
+            if (!IsConnected())
+            {
+                throw new Exception("MongoConnector is not currently connected");
+            }
+
+            try
+            {
+                ObjectId objectId = (ObjectId)doc["_id"];
+                var filter = Builders<BsonDocument>.Filter.Eq("_id", objectId);
+                var db = client.GetDatabase(dbName);
+                var devices = db.GetCollection<BsonDocument>(deviceCollection);
+                var result = devices.ReplaceOne(filter, doc);
+                return result.ModifiedCount > 0;
```

Comment:
Likewise, we could return the updated device here instead?

Reply:
I like this idea too, makes the return types more consistent

Reply:
I don't think its really necessary. The ui would already have the updated state and I don't think the rest of the api would need to use that object for anything.

Reply:
I don't think it is necessary either, since you would already have a copy of the document itself in memory in order to call this method in the first place. This way, by returning a boolean, we can keep sanity checking very easy and only have to do "if(result)". In essence, it would be redundant to return a copy of the document again.

Reply:
I think it's good for if we need to verify anything at the very least. I don't see too much harm in it.

Reply:
I agree, doesn't add any harm. We can use `if(result != NULL)`, that doesn't add any additional code. Thinking ahead to what BEIMA could be, what if we want undo functionality if a user deletes an entry they didn't mean to? What about keep a session log for a user and having to track all the entries a user deleted? We can do more with an object than we can with a boolean. Just some ideas

Reply:
Fixed d12ac92